### PR TITLE
Fix TAG_String

### PR DESCRIPTION
**Pull request by gentlegiantJGC**
## Problem
It was brought to my attention that Java uses a modified version of utf-8 while Bedrock uses standard utf-8 however it also uses the TAG_String class for arbitrary bytes and has also been known to store corrupted utf-8 strings (sliced in the middle of a byte)

This pull request should hopefully solve all of those issues.

All the cases are documented in this bug report https://github.com/Amulet-Team/Amulet-Map-Editor/issues/707

## Solution Part 1
The solution to the first half of this problem is to make the string decoder and encoder configurable.
This has been achieved by adding an input to the encode and decode methods that takes a function to do the encoding or decoding.

## Solution Part 2
The second half of the issue is the corrupted utf-8 cases.
This has been handled in the string encoding and decoding functions by adding an error handler that behaves like the backslashreplace error handler but instead of escaping the byte to `\xhh` it is escaped to `␛xhh` which is less likely to appear in normal text. After encoding with utf-8, the bytes are replaced with the original byte.

## Previous Solution
The previous solution was to store everything as the raw bytes but this had unforeseen side effects so this behaviour has been reverted and replaced with the escaping solution.

## Versioning
The py_bytes property has been removed. I think we should make this a 2.0 release due to the removal of functionality.

## Future
I will look into porting this behaviour to the rewritten cython library.

The cython rewrite is a bit different API wise to the V1 API. I think it would be wise port the V1 API to the rewrite and move the incompatible new behaviour behind a switch. We should mark the old behaviour as depreciated and give future warnings about features that change.
This will give programmers and us time to change and work out what code needs changing.
Once we feel we are ready we can flip the switch and enable the new behaviour.

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1 +1,2 @@
 numpy>=1.17.4
+mutf8~=1.0
```
Comment (Podshot) [inline requirements.txt:2]: It seems like mutf8 isn't imported/used anywhere, is this just an artifact before you went with the current implementation?

Comment (gentlegiantJGC) [reply requirements.txt:2]: That library exposes functions to encode and decode the modified utf-8 format.
It isn't directly used by the nbt library but the functions can be given to the library as encoder methods.
It made sense to include it here so dependents do not need to specify it.

**Review by Podshot — APPROVED** [review]
Looks all good to me!

**Comment by gentlegiantJGC** [comment]
I have ported this to the new version of the library.
Instead of merging this into the old library we should just include it in the new library.
I have made a quick patch to the old library to fix this issue.

**Pull request closed without merging**